src/include/common/int128_t.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <stdint.h>
+
+#include "common/api.h"
+#include "common/string_utils.h"
+
+namespace kuzu::common {
+
+struct int128_t;
+
+// System representation for int128_t.
+struct KUZU_API int128_t {
+    uint64_t low;
+    int64_t high;
+
+    int128_t() = default;
+    int128_t(int64_t value); // NOLINT: Allow implicit conversion from `int64_t`
+    constexpr int128_t(uint64_t low, int64_t high) : low(low), high(high) {}
+
+    constexpr int128_t(const int128_t&) = default;
+    constexpr int128_t(int128_t&&) = default;
+    int128_t& operator=(const int128_t&) = default;
+    int128_t& operator=(int128_t&&) = default;
+
+    // comparison operators
+    bool operator==(const int128_t& rhs) const;
+    bool operator!=(const int128_t& rhs) const;
+    bool operator>(const int128_t& rhs) const;
+    bool operator>=(const int128_t& rhs) const;
+    bool operator<(const int128_t& rhs) const;
+    bool operator<=(const int128_t& rhs) const;
+
+    // arithmetic operators
+    int128_t operator+(const int128_t& rhs) const;
+    int128_t operator-(const int128_t& rhs) const;
+    int128_t operator*(const int128_t& rhs) const;
+    int128_t operator/(const int128_t& rhs) const;
+    int128_t operator%(const int128_t& rhs) const;
+    int128_t operator-() const;
+
+    // inplace arithmetic operators
+    int128_t& operator+=(const int128_t& rhs);
+    int128_t& operator*=(const int128_t& rhs);
+
+    // boolean operators
+    explicit operator bool() const;
```

Comment:
Double check whether we need these operators or not.

Reply:
inplace mult is used in vector_list_function